sorc/ncep_post.fd/MDLFLD.f

```diff
@@ -3850,21 +3854,43 @@ SUBROUTINE MDLFLD
 !
          ENDIF
       ENDIF
-!     
-!           COMPUTE PBL HEIGHT BASED ON RICHARDSON NUMBER
-!     
+     
+!     -- COMPUTE/ASSIGN PBL HEIGHT ARRAY(S) --
+!
+!        J Kenyon (16 Jun 2025):
+!        Note that the PBL heights assigned below are subsequently used for
+!        other diagnostic calculations (e.g., 10-m wind gust, GTG fields).
+!        In some models (e.g., RAPR, FV3R), these PBL heights are only used
+!        internally within UPP, since a separate PBL height (calculated in
+!        the model) is provided in the GRIB2 output. Refer also to comments
+!        in CALPBL.f
+     
             IF ( (IGET(289)>0) .OR. (IGET(389)>0) .OR. (IGET(454)>0)   &
             .OR. (IGET(245)>0)  .or. IGET(464)>0 .or. IGET(467)>0  &
             .or. IGET(470)>0 .or. IGET(476)>0) THEN
-! should only compute pblri if pblh from model is not computed based on Ri 
-! post does not yet read pbl scheme used by model.  Will do this soon
-! For now, compute PBLRI for non GFS models.
+
+              !-- Regardless of model, assign / calculate PBLRI (PBL height based on Richardson number)
               IF(MODELNAME  ==  'GFS')THEN
                 PBLRI=PBLH
               ELSE
-               CALL CALPBL(PBLRI)
+                CALL CALPBL(PBLRI,'RI')
+              END IF
+              
+              !-- Additionally, for RAPR and FV3 only, assign PBLTHV (PBL height based on 
+              !   virtual potential temperature). For these models, PBLTHV is used for the
+              !   wind-gust diagnostic.
+              IF(((MODELNAME  ==  'RAPR').AND.(SUBMODELNAME /= 'MPAS'))  &
+                                         .OR. (MODELNAME  ==  'FV3R')) THEN
+                ! For older RAPR applications (with WRF-ARW) and FV3 applications: 
+                ! PBLHGUST is calculated in the associated INITPOST* routine;
+                ! simply pass PBLHGUST to PBLTHV
+                PBLTHV=PBLHGUST
```

Comment:
@jaymes-kenyon I suggest updating INITPOST_NETCDF.f to remove PBLHGUST calculation and use CALPBL for modelname 'FV3',  so that your new enhancement can benefit inline post in UFS-WM.